internal/controllers/addon/phase_ensure_subscription.go

```diff
@@ -96,6 +96,8 @@ func (r *olmReconciler) ensureSubscription(
 		return resultRetry, client.ObjectKey{}, nil
 	}
 
+	r.observedSubscription = *observedSubscription
```

Comment:
nit: out of an habit, it stresses me out when I see a pointer dereferenced without it being checked if it is nil before. checking above, where the object is created (in reconcilesubscription), it cannot be nil, unless error case, but still, maybe a comment here would help (for the sake of clarity), or nil case could be checked earlier, to be belt and braces.

Reply:
@gvanderpotte Thanks, that's just right, I forgot to check for nil.

Reply:
line 71: `observedSubscription, err := r.reconcileSubscription(ctx, desiredSubscription)`
observedSuscription is not nil if reconcileSubscription either returns a non-nil subscription pointer or a non-nil error.

Reply:
And that's the case: https://github.com/pepedocs/addon-operator/blob/fe70c681ed8bbc9da4ede8d8234600f93dbd5f33/internal/controllers/addon/phase_ensure_subscription.go#L107

Reply:
I added a nil check. For 2 reasons;

1. It's simpler to not assume about `reconcileSubscription`'s implementation.
2. There's still a possibility that observedSubscription will be nil (e.g. future code change) although unlikely. If this happens a nil check is required.